tests/src/test/java/org/eclipse/hono/tests/registry/TenantHttpIT.java

```diff
@@ -265,35 +262,25 @@ public void testGetAddedTenant(final VertxTestContext context)  {
     @Test
     public void testAddTenantFailsForMalformedTrustConfiguration(final VertxTestContext context) {
 
-        final JsonObject trustConfig = new JsonObject()
-                .put(TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, "test-dn")
-                .put(TenantConstants.FIELD_PAYLOAD_CERT, "NotBased64Encoded");
-        final JsonObject requestBody = buildTenantPayload()
-                .put(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA, trustConfig);
-        registry.addTenant(tenantId, requestBody, "application/json", HttpURLConnection.HTTP_BAD_REQUEST)
-        .setHandler(context.completing());
+        final Tenant requestBody = Tenants.createTenantForTrustAnchor("test-dn", "NotBased64Encoded".getBytes(), "RSA");
+
+        registry.addTenant(tenantId, requestBody, "application/json", HttpURLConnection.HTTP_BAD_REQUEST).setHandler(context.completing());
     }
 
     /**
-     * Creates a tenant object for a tenantId.
+     * Creates a tenant payload.
      * <p>
-     * The tenant object created contains configurations for the http and the mqtt adapter.
+     * The tenant payload contains configurations for the http, mqtt and a custom adapter.
      *
      * @return The tenant object.
      */
-    private static JsonObject buildTenantPayload() {
-
-        final JsonArray adapterConfigs = new JsonArray();
-        adapterConfigs.add(TenantObject.newAdapterConfig(Constants.PROTOCOL_ADAPTER_TYPE_HTTP, true)
-                .put(TenantConstants.FIELD_ADAPTERS_DEVICE_AUTHENTICATION_REQUIRED, true));
-        adapterConfigs.add(TenantObject.newAdapterConfig(Constants.PROTOCOL_ADAPTER_TYPE_MQTT, true)
-                .put(TenantConstants.FIELD_ADAPTERS_DEVICE_AUTHENTICATION_REQUIRED, true));
-        adapterConfigs.add(TenantObject.newAdapterConfig("custom", false)
-                .put(RegistryManagementConstants.FIELD_EXT, new JsonObject()
-                        .put("options", new JsonObject().put("maxInstances", 4))));
-        final TenantObject tenantPayload = TenantObject.from(true)
-                .setProperty(RegistryManagementConstants.FIELD_EXT, new JsonObject().put("plan", "unlimited"))
-                .setAdapterConfigurations(adapterConfigs);
-        return JsonObject.mapFrom(tenantPayload);
+    private static Tenant buildTenantPayload() {
+        final Tenant tenant = new Tenant();
+        tenant.setExtensions(new JsonObject().put("plan", "unlimited").getMap());
+        Tenants.setAdapterEnabled(tenant, Constants.PROTOCOL_ADAPTER_TYPE_HTTP, true, true);
```

Comment:
again, I think that the *setAdapterEnabled* method should be replaced by `Tenant.addAdapterConfiguration` ...

Reply:
please see my comment above